File: 寻找好区间.py
n, k = map(int, input().split())
ans, right = 0, 1
pre = [0 for _ in range(n + 1)]
s = list(input().split())
for i in range(1, n + 1):
    pre[i] = (s[i - 1] == "byl") + pre[i - 1]
for left in range(1, n + 1):
    while right <= n and pre[right] - pre[left - 1] < k:
        right += 1
    ans += n + 1 - right
print(ans)

File: 寻找好区间.cpp
#include <bits/stdc++.h>
using namespace std;

#define int long long

signed main() {
    int n, k, ans = 0, right = 1;
    string s;
    cin >> n >> k;
    vector<int> pre(n + 1);
    for (int i = 1; i <= n; ++i) {
        cin >> s;
        pre[i] = s == "byl";
        pre[i] += pre[i - 1];
    }
    for (int left = 1; left <= n; ++left) {
        while (pre[right] - pre[left - 1] < k and right <= n) {
            ++right;
        }
        ans += n + 1 - right;
    }
    cout << ans << endl;
    return 0;
}
